src/App.tsx

```diff
@@ -3,33 +3,56 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ query: event.target.value });
+  };
+
+  getVisibleMovies = (movies: Movie[]) => {
+    const { query } = this.state;
+
+    return (
+      movies.filter(
+        movie => movie.title.toLowerCase().includes(query.toLowerCase())
```

Comment:
no need to do `query.toLowerCase()` on every iteration, you can do it before filtering